### PR TITLE
Allow librarians to map all the tags in a sheet

### LINKED ISSUE #191: Add `" author"` to the end of all Identity tags instead of having dedicated Authorship tags

### PR DESCRIPTION
**Pull request by braxtonhall**
Resolves #191 

- [x] Update documentation

### CODE REVIEW

File: src/ts/content/adapters/tags/getTags.ts [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
 /**
  * The Meta Tag Sheet stores information about where tags are in the spreadsheet
  * Each row is of the form
- *   | SHEET | TOP_LEFT | WIDTH |
+ *   | SHEET | TOP_LEFT | WIDTH | MAPPER? | AS? |
  * where
  * SHEET    is the name of a sheet in the spreadsheet where tags live
  * TOP_LEFT is the top left most cell in the sheet the contains a tag
  * WIDTH    is how many columns contain tags
+ * MAPPER   is how tags should be mapped to after being consumed by BLT
```
Comment (adirar111) [inline src/ts/content/adapters/tags/getTags.ts:21]: so is the flow of this

* add the mapper to meta sheet
* BLL returns fetches unmapped tags as tags mapped according to mapper (they're still not updated in the sheet)
* they get updated in the sheet when users save stuff (?)

File: docs/librarian/tag-index.md [hunk 2]
```diff
@@ -47,3 +48,13 @@ Example sheet with Tags:
...
+
+### Tag Remapping
+
+A tag remapper is a string which transforms tags. A tag remapper _must_ contain `$TAG`.
+Tag remapping allows a Librarian to quickly add many tags that are transformations of existing tags.
+
+For example, given a `Colours` sheet containing the tags `Red`, `Green`, and `Blue`, and the tag remapper `$TAG author`,
+Better LibraryThing will artificially inflate the Tag Index with the new tags `Red author`, `Green author`, and `Blue author`.
+
+Created for VBL with the intention of Librarians remapping the `Indentity` sheet with `$TAG author`.
```
Comment (adirar111) [inline docs/librarian/tag-index.md:60]: typo?

File: src/ts/content/adapters/tags/getTags.ts (hunk 1 above)
Comment (adirar111) [reply src/ts/content/adapters/tags/getTags.ts:21]: i think i'm misunderstanding, but what happens when this happens twice? could it do the mapping operation twice? ex. `"Blue author author"`

**Review by adirar111 — APPROVED** [review]
had some questions for my own understanding

File: src/ts/content/adapters/tags/getTags.ts [hunk 3]
```diff
@@ -1,20 +1,25 @@
 import {makeCache} from "../../../common/util/cache";
 import {TagSearchOptions, TagTrees} from "./types";
-import Sheets, {Range} from "../sheets";
+import Sheets, {Range, ValueRange} from "../sheets";
 import {parseTree} from "./parseTags";
 import {incrementColumnBy} from "../sheets/util";
 
 declare const SPREADSHEET_ID: string; // Declared in webpack DefinePlugin
 
+type TagMapper = `${string}$TAG${string}`;
```
[the remark this answers is not in the document]
Comment (adirar111) [reply src/ts/content/adapters/tags/getTags.ts:9]: nvm it'll just prepend something before the tag

File: src/ts/content/adapters/tags/getTags.ts (hunk 1 above)
Comment (braxtonhall) [reply src/ts/content/adapters/tags/getTags.ts:21]: Hmm no I'll better document then. The original sheet doesn't get changed. It's like a materialized view in databases (I think -- it has been a while). So as far as the extension is concerned, there is an extra table, but it's created out of other source data.

With only `Blue`, `Red`, and `Green`, and a mapper, our tag validator/autocomplete will see Blue`, `Red`, `Green`, Blue author`, `Red author`, and `Green author`

**Pull request merged**